src/stores/history.ts

```diff
@@ -0,0 +1,29 @@
+import { defineStore } from "pinia";
+
+export const useHistoryStore = defineStore("history", () => {
+  const routes = ref<Array<{ fullPath: string; name: string }>>([]);
+
+  function addRoute(route: {
+    fullPath: string;
+    name: string | Symbol | null | undefined;
+  }) {
+    const { fullPath, name } = route;
+    routes.value.push({ fullPath, name: (name ?? "").toString() });
+  }
+
+  function goBack() {
+    routes.value.pop();
+    window.history.back();
+  }
+
+  const currentPageRoute = computed(
```

Comment:
destructure `routes.value` and give the array parts meaningful names so this becomes easier to understand?

Reply:
Not sure what exactly was meant to be too difficult  of a code :)
This exact method is now gone.
The only one left with at least some arithmetic returns a property from the second last element if an array has one.
To change it to destructure we will have to use .`reverse()` and idk if it really makes to code more readable. It doesn't make it shorter or faster (though of course you never know and it does not matter).

```
if (routes.value.length <= 1) return;
return routes.value[routes.value.length - 2].fullPath;

```
vs

```
const [currentRoute, previousRote, ...] = routes.value.reverse()
return previousRoute?.fullPath
```

I certainly can go for the second option in this case if you tell me this is the way, but in a generic scenario I would not do it without measuring performance (in a predictable language it would mean creating a reversed copy of a collection instead of fast access by index, though again in JS you never know).

I also would say that the function name is descriptive enough to give a hint on what its two lines of code do